policies/lib/tree/30_generic_methods/user_primary_group.cf

```diff
@@ -30,44 +30,40 @@
 bundle agent user_primary_group(login, primary_group)
 {
   vars:
-      "old_class_prefix"  string => canonify("user_primary_group_${login}");
-      "args"               slist => { "${login}", "${primary_group}" };
-      "report_param"      string => join("_", args);
-      "full_class_prefix" string => canonify("user_primary_group_${report_param}");
-      "class_prefix"      string => string_head("${full_class_prefix}", "1000");
+      "class_prefix"  string => canonify("user_primary_group_${login}");
       
   classes:
-      "user_exists" expression => userexists("${login}");
+      "user_exists"  expression => userexists("${login}");
       "group_exists" expression => groupexists("${primary_group}");
 
-      "args_defined" expression => "user_exists.group_exists";
+      "user_and_group_exist" expression => "user_exists.group_exists";
```

Comment:
used only once, I feel like this is unecessary

Reply:
used three times. espcially with a negation which should be translated to something rather complicated